source/tools/generateContractInterfaces.ts

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env node
+
+import * as fs from "async-file";
+import { Abi, AbiFunction, Primitive } from 'ethereum';
+import { CompilerOutputContracts, CompilerOutput } from "solc";
+import { Configuration } from './Configuration';
+
+doWork().then(() => {
+    process.exit();
+}).catch(error => {
+    console.log(error);
+    process.exit();
+});
+
+async function doWork(): Promise<void> {
+
+    const configuration = await Configuration.create();
+
+    const contractABIs: CompilerOutput = JSON.parse(await fs.readFile(configuration.contractOutputPath, "utf8"));
+
+    const fileContents: String = contractInterfacesTemplate(contractABIs.contracts);
+
+    await fs.writeFile(configuration.contractInterfacesOutputPath, fileContents);
+}
+
+function contractInterfacesTemplate(contracts: CompilerOutputContracts) {
```

Comment:
Style: The pattern I have been following is to have scripts be as short/simple as possible and putting all significant business logic into a separate file that can be called either from a script or from something else (such as a test or another library).  I also personally prefer class files rather than top level functions, though I don't have a strong argument for this if you have something particularly simple.

In this case, what _I_ would do is create a library class named `ContractInterfacesGenerator` that takes in a `Configuration` and a `ContractCompiler` as constructor parameters and then a single function on it called `generateContractInterfaces`.  This would allow us to provide mocks for `ContractCompiler` and `Configuration` should we want to author tests for this script at some point in the future, and it would follow the pattern the rest of our TS stuff follows (dependency injection with classes).  `doWork` above would then just instantiate `Configuration`, instantiate `ContractCompiler`, instantiate `ContractInterfacesGenerator` and then call `generateContractInterfaces`.

I would probably also have `generateContractInterfaces` return the generated string, and also write the file to disk (in case a future test author wants to validate the string).

We don't have an agreed upon style guide/pattern set for TypeScript, so all of this is up for discussion.  The above is just the pattern I have been following elsewhere, and in lieu of any discussion consistency is the current reigning champion that drives my feedback here.

Reply:
That makes sense. I like putting what we can into library files anyway so I'll do this.